### ISSUE TITLE
Async_replication context deadline exceeded, unable to Activate Tenant

### ISSUE THREAD

**Issue by harvetech** [weaviate/weaviate#6380]
### How to reproduce this bug?

Run weaviate on a geographically spread mixed Arch Kubernetes 1.30.

- Server Setup Information
- Weaviate Server Version: 1.27
- Deployment Method: Helm Chart
- Multi Node? 4
- Client Language and Version: Python, 4.9.3
- Multitenancy?: Yes

### What is the expected behavior?

It should just work. Maybe slow because of latencies, but the basic functionality should stay intact. 

### What is the actual behavior?

Hi all, I am trying to run Weaviate on Mixed Arch Kube ( Arm65 and Amd64). 3-Nodes and 3 Replicas. Nodes are geographically apart (2 Nodes at one place, 1 at another).But I am constantly getting these timeouts, only for a Tenant which has almost 3GB data.

`{“action”:“async_replication”,“build_git_commit”:“64457c2”,“build_go_version”:“go1.22.9”,“build_image_tag”:“v1.27.2”,“build_wv_version”:“1.27.2”,“class_name”:“Question”,“hashbeat_iteration”:1679,“level”:“warning”,“msg”:“hashbeat iteration failed: collecting differences: "10.233.84.228:7001": connect: Post "[http://10.233.84.228:7001/replicas/indices/Question/shards/tenantA/objects/hashtree/0?schema_version=0\](http://10.233.84.228:7001/replicas/indices/Question/shards/tenantA/objects/hashtree/0?schema_version=0%5C)”: context deadline exceeded",“shard_name”:“tenantA”,“time”:“2024-11-18T06:48:56Z”}`

### Supporting information

ativating client, sometimes works, sometime it times out, usually problem appears after offloading to s3 and then setting the client active again.
![image](https://github.com/user-attachments/assets/71fdb275-a383-42f3-913e-680658a7da17)

### Server Version

1.27

### Weaviate Setup

Multi-Node Cluster

### Nodes count

4

### Code of Conduct

- [X] I have read and agree to the Weaviate's [Contributor Guide](https://weaviate.io/developers/contributor-guide) and [Code of Conduct](https://weaviate.io/service/code-of-conduct)

**Comment by harvetech** [comment]
Update: 
After debugging further, I found out that the settings of GOGC and GOMEMLIMIT were not optimal. After setting these values properly, I don't see this error anymore. However, I still can't activate a client, it timeout or give a weird error. The nodes are not at all under any significant load. 

How to reproduce:

1. Create collection, add vectors, offload to s3
2. Try to activate the collection.  

The following error I see. 

> UnexpectedStatusCodeError: Collection tenants may not have been updated properly for Questions! Unexpected status code: 422, with response body: {'error': [{'message': 'rpc error: code = Internal desc = updating schema: TYPE_UPDATE_TENANT: class not found'}]}.

At the same time I see these logs in the pods 

> {"build_git_commit":"64457c2","build_go_version":"go1.22.9","build_image_tag":"v1.27.2","build_wv_version":"1.27.2","class":"Questions","level":"debug","msg":"server.execute","time":"2024-11-18T16:33:29Z","type":"TYPE_UPDATE_TENANT"}
> {"build_git_commit":"64457c2","build_go_version":"go1.22.9","build_image_tag":"v1.27.2","build_wv_version":"1.27.2","cmd_class":"Questions","cmd_schema_only":false,"cmd_type":17,"cmd_type_name":"TYPE_UPDATE_TENANT","level":"debug","log_index":440,"log_name":"LogCommand","log_type":0,"msg":"server.apply","time":"2024-11-18T16:33:29Z"}
> {"build_git_commit":"64457c2","build_go_version":"go1.22.9","build_image_tag":"v1.27.2","build_wv_version":"1.27.2","cmd_class":"Questions","cmd_type":17,"cmd_type_name":"TYPE_UPDATE_TENANT","error":"updating schema: TYPE_UPDATE_TENANT: class not found","level":"error","log_index":440,"log_name":"LogCommand","log_type":0,"msg":"apply command","time":"2024-11-18T16:33:29Z"}


I think this is coming from the python code which is used to activate and deactivate a client. It is somehow corrupting the collection object and once it's corrupted you can't activate it anymore.

**Comment by harvetech** [comment]
Update: 

```
test_collection = client.collections.create(
        "test",
        multi_tenancy_config=Configure.multi_tenancy(True),
        vectorizer_config=wvc.config.Configure.Vectorizer.none(),
        replication_config=Configure.replication(
            factor=3,
            async_enabled=True,
        ),
        #sharding_config=Configure.sharding(
        #    virtual_per_physical=128,
        #    desired_count=4,
        #    actual_count=4,
        #    desired_virtual_count=128,
        #),
        vector_index_config=Configure.VectorIndex.hnsw(
            quantizer=Configure.VectorIndex.Quantizer.bq(),
            ef_construction=300,
            distance_metric=VectorDistances.COSINE,
            filter_strategy=VectorFilterStrategy.SWEEPING,  # or ACORN (Available from Weaviate v1.27.0)
            vector_cache_max_objects=1000000
        )
    )
```

When I try to add a new collection I get timeout and in the pod logs I see this.

> {"build_git_commit":"64457c2","build_go_version":"go1.22.9","build_image_tag":"v1.27.2","build_wv_version":"1.27.2","cmd_class":"Test","cmd_schema_only":false,"cmd_type":1,"cmd_type_name":"TYPE_ADD_CLASS","level":"debug","log_index":505,"log_name":"LogCommand","log_type":0,"msg":"server.apply","time":"2024-11-19T07:37:46Z"}
> 
> {"build_git_commit":"64457c2","build_go_version":"go1.22.9","build_image_tag":"v1.27.2","build_wv_version":"1.27.2","cmd_class":"Test","cmd_type":1,"cmd_type_name":"TYPE_ADD_CLASS","error":"updating schema: TYPE_ADD_CLASS: class already exists","level":"error","log_index":507,"log_name":"LogCommand","log_type":0,"msg":"apply command","time":"2024-11-19T07:48:46Z"}

**Comment by harvetech** [comment]
Update:

After I restarted the pods, it seems to work again. It looks like python package has the possibility to corrupt the pods. The main problem is offloading to s3 and then activating the client is the problematic part. It's very unstable in my current setup. If offloading works, the activation won't. I can see that out of 3 pods, 2 pods got the data from s3 but the one pod never. So even if the tenant activation code works but I cannot query the collection, nor I can add some vectors and all the operations on the collections are timed out. 

as you can see the weaviate-0 is still in the offload folder

![image](https://github.com/user-attachments/assets/b852775a-edfa-4c11-9d2c-ad317f41a838)

and this code 

```
for k in range(10):

    query_vector = np.random.uniform(-1, 1, size=1536).astype(np.float32)

    response = questiion_tenantA.query.near_vector(
        near_vector=query_vector,
        limit=50,
        return_metadata=wvc.query.MetadataQuery(certainty=True)
    )
    #print(response)
    print(k)
```
results in this error 

> WeaviateQueryError: Query call with protocol GRPC search failed with message <AioRpcError of RPC that terminated with:
> 	status = StatusCode.DEADLINE_EXCEEDED
> 	details = "Deadline Exceeded"
> 	debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"Deadline Exceeded", grpc_status:4, created_time:"2024-11-19T10:11:00.55771786+01:00"}"
> >.

**Comment by jeroiraz** [comment]
Update: I'm able to reproduce the timeout situation and working on optimizing the implementation

**Comment by I321065** [comment]
Hi @jeroiraz , 
  I also face a similar issue, please check if it is caused by same reason:
  
![image](https://github.com/user-attachments/assets/4e56a814-9117-443b-9083-b474c017719e)

version: 1.27.2
data replicas: 3
node number: 3

**Comment by jeroiraz** [comment]
> Hi @jeroiraz , I also face a similar issue, please check if it is caused by same reason:
> 
> ![image](https://private-user-images.githubusercontent.com/16135063/392254999-4e56a814-9117-443b-9083-b474c017719e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMyOTUxODksIm5iZiI6MTczMzI5NDg4OSwicGF0aCI6Ii8xNjEzNTA2My8zOTIyNTQ5OTktNGU1NmE4MTQtOTExNy00NDNiLTkwODMtYjQ3NGMwMTc3MTllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA0VDA2NDgwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzYjVmMzg3MmNlMTU2MmIyMmU3ZWNmY2Y5M2M2OWI4ZDMzZTNmOTc0N2ViZDVhMmNkNmQxN2YxMmM2MjY5OTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.w4j-IdwTdOlXeOeoiCpDBY9_FK1JeO5O8u9VEwIdyDM)
> 
> version: 1.27.2 data replicas: 3 node number: 3

Hi @I321065 yes, I think it's the same reason, thanks for reaching out. We're working on it this week

**Comment by jeroiraz** [comment]
This issue should be solved as of v1.26.13 release (https://github.com/weaviate/weaviate/releases/tag/v1.26.13)

**Comment by lidija-jovanovska** [comment]
Hi! I'm having the same issue using version 1.28.2 and a three-node setup. 

What happened exactly was.
1) Weaviate was working normally
2) I used the python client (v. 4.10.2) to delete a collection
3) I got a `httpx.RemoteProtocolError: Server disconnected without sending a response.` error. Although, I could see similarly as @harvetech that the number of objects decreased to 0, so the request must have gone through.
4) Immediately after that Weaviate stopped working and I started seeing these errors:

`{"action":"async_replication","build_git_commit":"5a3991d","build_go_version":"go1.22.10","build_image_tag":"v1.28.2","build_wv_version":"1.28.2","class_name":"Match_embedding_model_2_8_0","hashbeat_iteration":2604,"level":"warning","msg":"hashbeat iteration failed: collecting differences: \"100.64.68.2:7001\": connect: Post \"http://100.64.68.2:7001/replicas/indices/Match_embedding_model_2_8_0/shards/ODLwlrW92yMM/objects/hashtree/0?schema_version=0\": context deadline exceeded","shard_name":"ODLwlrW92yMM","time":"2025-01-07T08:43:39Z"}`

After restarting Weaviate, the database behaves as expected again. To ensure that this is the reason for the malfunctioning, I ran the process twice. Any help would be appreciated @jeroiraz
